presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -383,6 +393,16 @@ public ActualProperties visitJoin(JoinNode node, List<ActualProperties> inputPro
                     constants.putAll(probeProperties.getConstants());
                     constants.putAll(buildProperties.getConstants());
 
+                    if (node.isCrossJoin()) {
+                        // Cross join preserves only constants from probe and build sides.
+                        // Cross join doesn't preserve sorting or grouping local properties on either side.
+                        return ActualProperties.builderFrom(probeProperties)
+                                .local(ImmutableList.of())
+                                .constants(constants)
+                                .unordered(unordered)
```

Comment:
this is no longer needed

Reply:
@sopel39 Karol, just to clarify, are you pointing out that with empty local properties `.unordered(unordered)` is redundant? I missed that. Thanks. Will update.